### PR TITLE
Better eureka-based discovery integration

### PR DESCRIPTION
**Pull request by jvmlet**
1. Spring cloud version promoted  to  Brixton.M3
2. Added EurekaApplicationDiscoveryListener that properly propagates service-configured health and management URLs to Application

Should address #154

### CODE REVIEW

**Comment by jvmlet** [comment]
@joshiste , please merge.

**Comment by jvmlet** [comment]
@joshiste 
Please merge only 
*https://github.com/jvmlet/spring-boot-admin/commit/62b7b40be56f0574502cf928633fdc9207100fd0
and
*https://github.com/jvmlet/spring-boot-admin/commit/b21f756da8c8928df475861c996af4734050f4b2
commits
Sorry

**Comment by joshiste** [comment]
You better would have done a rebase.... And forced push

**Comment by joshiste** [comment]
I startend to die into the changes. Unfortunatly the upgrade to a current Brixton build snapshot is not a no-brainer and will take me sind days.

**Comment by jvmlet** [comment]
That's why I've upgraded to Brixton.M3, not to the latest M5 which introduced major non backward compatible API changes.

**Comment by joshiste** [comment]
On hold, until Spring Cloud Brixton has been released.

File: spring-boot-admin-server/src/test/java/de/codecentric/boot/admin/discovery/eureka/EurekaDiscoveryListenerTest.java [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+/**
+ * Created by alexf on 11-Feb-16.
+ */
+
+public class EurekaDiscoveryListenerTest {
+
+    private EmbeddedWebApplicationContext eurekaServer;
+
+    @Before
+    public void setup() {
```
Comment (joshiste) [inline spring-boot-admin-server/src/test/java/de/codecentric/boot/admin/discovery/eureka/EurekaDiscoveryListenerTest.java]: Codacy Issue found: [You may have misspelled a JUnit framework method (setUp or tearDown)](https://www.codacy.com/app/joshiste/spring-boot-admin/file/2229709767/issues/source?bid=3175464&fileBranchId=3175465#l30)

File: spring-boot-admin-server/src/test/java/de/codecentric/boot/admin/controller/JournalControllerTest.java [hunk 2]
```diff
@@ -48,4 +50,15 @@ public void test_getJournal() {
 		ClientApplicationEvent event = history.iterator().next();
 		assertThat(event, sameInstance(emittedEvent));
 	}
+
+	@Test
```
Comment (joshiste) [inline spring-boot-admin-server/src/test/java/de/codecentric/boot/admin/controller/JournalControllerTest.java]: Codacy Issue found: [Method names should not contain underscores](https://www.codacy.com/app/joshiste/spring-boot-admin/file/2229709764/issues/source?bid=3175464&fileBranchId=3175465#l55)

**Comment by joshiste** [comment]
closed by 272633e1b16b88125d929868a6bcf5338c6363b3

**Pull request closed without merging**